### PR TITLE
race condition fix in accept loop - fixes #14

### LINKED ISSUE #14: 'race' condition.
If an error causes the listener to return early from `accept()` it will close the `acceptErrs` chan, but if the [goroutine here](https://github.com/jbenet/go-peerstream/blob/master/listener.go#L106) hasnt completed yet, then we are going to get the send on closed channel panic i see here: https://gist.github.com/whyrusleeping/c67a5b082f77c648c4b7

### PR DESCRIPTION
**Pull request by jbenet**

### CODE REVIEW

**Comment by jbenet** [comment]
@whyrusleeping CR pls

File: listener.go [hunk 1]
```diff
@@ -74,6 +75,8 @@ func ListenersWithGroup(g Group, ls []*Listener) []*Listener {
 // TODO: add rate limiting
 func (l *Listener) accept() {
 	defer l.teardown()
+	var wg sync.WaitGroup
+	defer wg.Wait()
```
Comment (whyrusleeping) [inline listener.go]: I'd comment here and say something like 'defers are calling in LIFO order, this will execute before `l.teardown()`'

**Comment by whyrusleeping** [comment]
one comment, tests fail, but otherwise LGTM

File: listener.go (hunk 1 above)
Comment (jbenet) [inline listener.go]: you're right. fixed in 1eb1668

**Comment by jbenet** [comment]
> tests fail

tests fail becuase it's running -race tests which are too intense apparently

**Pull request merged**